aggregator_core/src/datastore.rs

```diff
@@ -1046,7 +1056,9 @@ impl<C: Clock> Transaction<'_, C> {
             )?));
         }
 
-        Ok(Task::new(
+        // Don't validate a task on its way out. If it's in the database, we'll assume that it was
+        // constructed in a valid way.
+        Ok(Task::new_without_validation(
```

Comment:
Indeed. Taskprov tasks don't have any auth tokens or a collector HPKE config, which Task::validate considers invalid.

Reply:
An alternative could to track in the database whether a task was created by taskprov, then invoke separate validation logic. Curious on your thoughts on this--I was wary of mixing taskprov logic too much with normal logic, hence the approach of building a taskprov task onto the normal task abstraction as much as possible.

Reply:
I think you're right that errors here are unlikely, but validating data here is still nice because it can protect us against a botched database migration or some other funny thing happening.

I wonder if tracking the source of the task will become necessary anyway so that divviup-api can make good decisions. But in the near term, I'm wary of making schema changes unless we're sure we need them. How about a compromise: you've written two flavours of task validation routine, one for taskprov tasks and one for regular ones. We could check if either succeeds and error out otherwise. This way all we're doing is adding code, which is much easier to change than a DB schema. If at some point in the future we begin tracking task provenance anyway, we can revisit this validation logic.

Reply:
I've implemented this. I think it could probably use some unit tests around validation, but I'd like your input before I get too deep into that.